tests/test_auth.py

```diff
@@ -321,11 +379,113 @@ def test_custom_token(self, authtest):
+            'iat': int(time.time()) - 10000,
+            'exp': int(time.time()) - 3600
+        }),
+        'NoneCookie': None,
+        'EmptyCookie': '',
+        'BoolCookie': True,
+        'IntCookie': 1,
+        'ListCookie': [],
+        'EmptyDictCookie': {},
+        'NonEmptyDictCookie': {'a': 1},
+        'BadFormatCookie': 'foobar',
+        'IDToken': TEST_ID_TOKEN,
+    }
+
+    def setup_method(self):
+        _token_gen._request = testutils.MockRequest(200, MOCK_PUBLIC_CERTS)
+
+    @pytest.mark.parametrize('cookie', valid_cookies.values(), ids=list(valid_cookies))
+    def test_valid_cookie(self, authtest, cookie):
+        claims = authtest.verify_session_cookie(cookie)
+        assert claims['admin'] is True
+        assert claims['uid'] == claims['sub']
+
+    @pytest.mark.parametrize('cookie', valid_cookies.values(), ids=list(valid_cookies))
+    def test_valid_cookie_check_revoked(self, user_mgt_app, cookie):
+        _instrument_user_manager(user_mgt_app, 200, MOCK_GET_USER_RESPONSE)
+        claims = auth.verify_session_cookie(cookie, app=user_mgt_app, check_revoked=True)
+        assert claims['admin'] is True
+        assert claims['uid'] == claims['sub']
+
+    @pytest.mark.parametrize('cookie', valid_cookies.values(), ids=list(valid_cookies))
+    def test_revoked_cookie_check_revoked(self, user_mgt_app, cookie):
+        _instrument_user_manager(user_mgt_app, 200, MOCK_GET_USER_REVOKED_TOKENS_RESPONSE)
+
+        with pytest.raises(auth.AuthError) as excinfo:
+            auth.verify_session_cookie(cookie, app=user_mgt_app, check_revoked=True)
+
+        assert excinfo.value.code == 'SESSION_COOKIE_REVOKED'
+        assert str(excinfo.value) == 'The Firebase session cookie has been revoked.'
+
+    @pytest.mark.parametrize('cookie', valid_cookies.values(), ids=list(valid_cookies))
+    def test_revoked_cookie_does_not_check_revoked(self, user_mgt_app, cookie):
+        _instrument_user_manager(user_mgt_app, 200, MOCK_GET_USER_REVOKED_TOKENS_RESPONSE)
+        claims = auth.verify_session_cookie(cookie, app=user_mgt_app, check_revoked=False)
+        assert claims['admin'] is True
+        assert claims['uid'] == claims['sub']
+
+    @pytest.mark.parametrize('cookie', invalid_cookies.values(), ids=list(invalid_cookies))
+    def test_invalid_cookie(self, authtest, cookie):
+        with pytest.raises(ValueError):
+            authtest.verify_session_cookie(cookie)
+
+    def test_project_id_option(self):
+        app = firebase_admin.initialize_app(
+            testutils.MockCredential(), options={'projectId': 'mock-project-id'}, name='myApp')
+        try:
+            claims = auth.verify_session_cookie(TEST_SESSION_COOKIE, app=app)
+            assert claims['admin'] is True
+            assert claims['uid'] == claims['sub']
+        finally:
+            firebase_admin.delete_app(app)
+
+    @pytest.mark.parametrize('env_var_app', [{'GCLOUD_PROJECT': 'mock-project-id'}], indirect=True)
```

Comment:
Here and below, why parametrize for a single test?

Reply:
They are passed indirectly (`indirect=True`) to a fixture.